bpfman/src/types.rs

```diff
@@ -1144,10 +1144,14 @@ impl ProgramData {
 
     pub(crate) fn set_kernel_info(&mut self, prog: &AyaProgInfo) -> Result<(), BpfmanError> {
         self.set_id(prog.id())?;
-        self.set_kernel_name(
-            prog.name_as_str()
-                .expect("Program name is not valid unicode"),
-        )?;
+        if prog.name().is_empty() {
+            self.set_kernel_name("None")?;
+        } else {
+            self.set_kernel_name(
+                prog.name_as_str()
+                    .expect("Program name is not valid unicode"),
+            )?;
+        }
```

Comment:
```suggestion
        self.set_kernel_name(prog.name_as_str().unwrap_or("None"))?;
```

We've been working on handling errors better rather than doing things that can cause a panic like `expect()`.  In this case, I think it would be better to set the name to "None" for any case that `set_kernel_name()` returns `None` rather than panic.

Reply:
Thank you for your corrections. bpfman is my first project learning Rust, so I'm not very familiar with some Rust interfaces yet. :)